libs/core/langchain_core/messages/base.py

```diff
@@ -95,9 +95,13 @@ def merge_content(
         else:
             return_list: List[Union[str, Dict]] = [first_content]
             return return_list + second_content
-    # If both are lists, merge them naively
     elif isinstance(second_content, List):
-        return first_content + second_content
+        # If both are lists
+        merged_list = merge_lists(first_content, second_content)
+        # Satisfy type-checker
+        if merged_list is None:
+            raise Exception("Failed to merge message chunk contents.")
```

Comment:
(nit) if we know this will be a list can also import `cast` from `typing` and do `return cast(list, merged_list)`

Reply:
Done